crates/bevy_ecs/src/world/pointer.rs

```diff
@@ -39,6 +39,18 @@ impl<'a, T: core::fmt::Debug> core::fmt::Debug for Mut<'a, T> {
     }
 }
 
+impl<'w, T> AsRef<T> for Mut<'w, T> {
+    fn as_ref(&self) -> &T {
+        &*self
+    }
+}
+
+impl<'w, T> AsMut<T> for Mut<'w, T> {
+    fn as_mut(&mut self) -> &mut T {
+        &mut *self
```

Comment:
Ditto here for the marking changed status.

Reply:
I think the `*` will call `DerefMut`'s `deref_mut`, which will do that automatically, but I've made that explicit now

Reply:
Oh if that's the case with the `DerefMut` trait, then I'm happy leaving it as is.